src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -76,6 +86,16 @@ internal bool IsPropertyEnumerable()
             return false;
         }
 
+        internal bool IsPropertyADictionary()
+        {
+            if (JsonPropertyInfo != null)
+            {
+                return JsonPropertyInfo.ClassType == ClassType.Dictionary;
```

Comment:
Make sure to run code coverage to check for dead code or see where code coverage is missing in tests.